src/vmm/src/resources.rs

```diff
@@ -237,6 +237,21 @@ impl VmResources {
         }
     }
 
+    /// Assign the custom CPU template to VM resources to
+    /// be used for configuring the guest vCPUs.
+    pub fn update_custom_cpu_template(
+        &mut self,
+        custom_cpu_template: CustomCpuTemplateType,
+    ) -> std::result::Result<(), GuestConfigurationError> {
+        if self.vm_config.cpu_template != CpuFeaturesTemplate::None {
```

Comment:
`set_custom_cpu_template` is reasonable. Will change.

----------------

> Ideally we want to give the user the possibility to go back and forth between a static and a custom one.

I agree. They should be able to go back and forth. A simple and clear approach as follows:
* If a static CPU template is set, but actually want to use a custom template, they will be able to set the static template to `CpuFeaturesTemplate::None`, and then configure the custom template.
* If a custom template is set, they can "unset" it with an empty template, and then configure the static template.

There are small variations to this we could introduce, like if you configure a static template, it automatically sets the static template to "None". I think requiring the user to do "unset step" is much clearer. This will be easy enough to customize though.